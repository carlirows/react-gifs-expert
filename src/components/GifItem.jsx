

export const GifItem = ({gif}) => {
  return (
    <div className="card">
        <img src={gif.url} alt={gif.title} />
        <p key={gif.id}>{gif.title}</p>
    </div>
  )
}

