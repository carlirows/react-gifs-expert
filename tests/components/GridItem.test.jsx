import { render, screen } from '@testing-library/react';
import { GifItem } from '../../src/components/GifItem';


describe('test en GridItem Component', () => { 
    const title='Titulander'
    const url='https://www.elinmortal.com/'
    test('deberia hacer match copn el snapshot', () => { 
        const {container} = render(<GifItem title={title} url={url}/>);
        expect(container).toMatchSnapshot();
     })
    test('deberia mostrar la imagen con el url y alt indicado', () => { 
        const {container} = render(<GifItem title={title} url={url}/>);
        //screen.debug(): //por si necesito ver el sujeto de pruebas
        //expect(screen.getByRole('img').src).toBe(url)
        //expect(screen.getByRole('img').alt).toBe(title)
        const { src, alt } = screen.getByRole('img') 
        expect(src).toBe(url)
        expect(alt).toBe(title)
     })
    test('deberia mostrar el titulo en el componente', () => { 
        render(<GifItem title={title} url={url}/>);
        expect(screen.getByText=(title)).toBeTruthy()
        
     })
 })