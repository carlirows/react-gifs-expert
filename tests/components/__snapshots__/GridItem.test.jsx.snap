// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test en GridItem Component deberia hacer match copn el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Titulander"
      src="https://www.elinmortal.com/"
    />
    <p>
      Titulander
    </p>
  </div>
</div>
`;
